fuse/src/main/java/tachyon/fuse/TachyonFuse.java

```diff
@@ -0,0 +1,166 @@
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.HelpFormatter;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import tachyon.Constants;
+import tachyon.client.ClientContext;
+import tachyon.client.file.TachyonFileSystem;
+import tachyon.client.file.TachyonFileSystem.TachyonFileSystemFactory;
+import tachyon.conf.TachyonConf;
+
+/**
+ * Main entry point to Tachyon-FUSE
+ */
+public final class TachyonFuse {
+  private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
+  private static TachyonConf sTachyonConf;
+
+  /**
+   * Running this class will mount the file system according to
+   * the options passed to this function {@link #parseOptions(String[])}.
+   * The user-space fuse application will stay on the foreground and keep
+   * the file system mounted. The user can unmount the file system by
+   * gracefully killing (SIGINT) the process.
+   */
+  public static void main(String[] args) {
+    sTachyonConf = ClientContext.getConf();
+    final TachyonFuseOptions opts = parseOptions(args);
+    if (opts == null) {
+      System.exit(1);
+    }
+
+    final TachyonFileSystem tfs = TachyonFileSystemFactory.get();
+    final TachyonFuseFs fs = new TachyonFuseFs(sTachyonConf, tfs, opts);
+    final List<String> fuseOpts = opts.getFuseOpts();
+    // Force direct_io in FUSE: writes and reads bypass the kernel page
+    // cache and go directly to tachyon. This avoids extra memory copies
+    // in the write path.
+    fuseOpts.add("-odirect_io");
+
+    try {
+      fs.mount(Paths.get(opts.getMountPoint()), true, opts.isDebug(),
+          fuseOpts.toArray(new String[0]));
+    } finally {
+      fs.umount();
+    }
+  }
+
+  /**
+   * Parses CLI options
+   * @param args CLI args
+   * @return Tachyon-FUSE configuration options
+   */
+  private static TachyonFuseOptions parseOptions(String[] args) {
+    final Options opts = new Options();
+    final Option mntPoint = Option.builder("m")
+        .hasArg()
+        .required(false)
+        .longOpt("mount-point")
+        .desc("Desired local mount point for tachyon-fuse.")
+        .build();
+
+    final Option tachyonRoot = Option.builder("r")
+        .hasArg()
+        .required(false)
+        .longOpt("tachyon-root")
+        .desc("Path within tachyon that will be used as the root of the FUSE mount "
+            + "(e.g., /users/foo; defaults to /)")
+        .build();
+
+    final Option help = Option.builder("h")
+        .required(false)
+        .desc("Print this help")
+        .build();
+
+    final Option fuseOption = Option.builder("o")
+        .valueSeparator(',')
+        .required(false)
+        .hasArgs()
+        .desc("FUSE mount options")
+        .build();
```

Comment:
I find this style much more readable and less error-prone. I see that the "old" style is used in other few places (tachyon.yarn.Client, tachyon.master.journal.JournalTool, tachyon.examples.JournalCrashTest). If consistency of style is a priority, my opinion is that the other classes are changed. If you instead have other reasons for not using this style, I am OK to change this. 

In both cases, I can open a related Jira to fix this.